core/src/main/groovy/noe/rhel/server/httpd/HttpdBaseOS.groovy

```diff
@@ -0,0 +1,119 @@
+
+  HttpdBaseOS(String basedir, version) {
+    super(basedir, version)
+    this.serviceName = "httpd"
+    this.basedir = "/etc/${serviceName}"
+
+    // please initialize in setDefault() method
+    setDefault()
+    logDirs.each { JBFile.makeAccessible(new File(getServerRoot() + it)) }
+  }
+
+  void setDefault() {
+    super.setDefault()
+    this.configDirs = ['/conf', '/conf.d']
+    if (version >= new Version("2.4")) {
+      this.configDirs.add('/conf.modules.d')
+    }
+    this.modulesDir = this.basedir + "/modules"
+    this.logDirs = ['/logs']
+    // what path will be taken as relative for star/stop cmd
+    this.binPath = '/'
+    if (platform.isRHEL6() || platform.isRHEL5() || platform.isRHEL4()) {
+      this.start = ['service', serviceName, 'start']  // starting as service
+      this.stop = "service ${serviceName} stop"  // stopping as service
+    } else {
+      this.start = ['systemctl', 'start', serviceName]
+      this.stop = ['systemctl', 'stop', serviceName]
+    }
+    this.apachectl = ['/usr/sbin/apachectl']
+    this.deploymentPath = "/var/www/html"
+    this.confDeploymentPath = this.basedir + '/conf.d'
+    this.cgiDeploymentPath = "/var/www/cgi-bin"
+    this.modClusterCacheDir = "/var/cache/mod_cluster"
+    this.opensslPath = 'openssl' // openssl is on $PATH on RHEL by OS installation
+    this.abPath = "/usr/bin/ab"
+    this.htdbmPath = "/usr/bin/htdbm"
+    this.htpasswdPath = "/usr/bin/htpasswd"
+    this.httxt2dbmPath = "/usr/bin/httxt2dbm"
+    this.logresolvePath = "/usr/bin/logresolve"
+
+    // ! please add new paths as absolute paths
+  }
+
+  File getPidFile() {
+    return new File("/var/run/${serviceName}/${serviceName}.pid")
+  }
+
+  Integer extractPid() {
+    def pidAsStr = new ByteArrayOutputStream()
+    if (Cmd.executeCommandRedirectIO("sudo cat ${getPidFile()}", null, null, pidAsStr, System.err) == 0) {
+      pidAsStr = pidAsStr.toString()
+      pidAsStr = pidAsStr.trim()
+      pidAsStr = pidAsStr.replaceAll('"', '')
+      pid = Integer.valueOf(pidAsStr)
+
+      return pid
+    } else {
+      log.debug("PID file is not accessible. B start(ut continuing ...")
+      return null
+    }
+  }
+
+  /**
+   * Stop the server
+   */
+  void start(Map conf = [:]) {
+    super.start(conf)
+    Library.letsSleep(5000)
+  }
+
+  /**
+   * Stop the server
+   */
+  long stop(Map conf = [:]) {
+    long stopTime = super.stop(conf)
+    Library.letsSleep(3000)
+    return stopTime
+  }
+
+  void setRequestedSELinuxContext(String context) {
+    this.start = ['runcon', '-t', context, this.start]
+  }
+
+  void backupConfs() {
+    configDirs.each { dir ->
+      def srcDir = getServerRoot() + dir
+      def bkpDir = srcDir + ".bkp"
+
+      Cmd.executeCommand("sudo cp --recursive --preserve --dereference $srcDir $bkpDir", new File("."))
+    }
+
+    configDirs.each { JBFile.makeAccessible(new File(getServerRoot() + it)) }
+  }
+
+  void restoreConfs() {
+    configDirs.each { dir ->
```

Comment:
same as above

Reply:
next commit

Reply:
Why do we have here this code?

Reply:
I found out some superfluous configuration files present after test execution.